include/rcutils/filesystem.h

```diff
@@ -144,6 +144,25 @@ rcutils_to_native_path(
   const char * path,
   rcutils_allocator_t allocator);
 
+/// Create the specified directory.
+/**
+ * This function creates an absolutely-specified directory.  If any of the
+ * intermediate directories do not exist, this function will return False.
+ * If the abs_path already exists, and is a directory, this function will
+ * return True.
+ *
+ * \param[in] abs_path
+ * \param[in] allocator
+ * \return bool True if making the directory was successful, False otherwise
+ *              False if path is NULL
+ *              False if path is empty
+ *              False if path is not absolute
+ *              False if any intermediate directories don't exist
+ */
+RCUTILS_PUBLIC
```

Comment:
Is is worth noting that this function is likely not thread safe?  I think most folks would assume that is the case, given the underlying call, but could probably bear repeating.

Reply:
Hm, interesting question.  It sort of depends on our definition of thread-safe.  It only uses stack variables and the parameter passed in, so there is no risk of memory corruption/memory races.  On the other hand, there is the directory race that `mkdir` is subject to (described in the openat documentation): http://man7.org/linux/man-pages/man2/open.2.html .  So from that perspective, it isn't thread-safe, but I'm not sure how we want to describe it.

Reply:
I think leaving it at "This function is not thread safe" is probably suitable.  If someone is deeply concerned about it, then can consult the underlying code and documentation?

Reply:
I ended up adding that with a minor note to go look at the openat documentation for more information.